### PR TITLE
[Lang] Add append attribute to dynamic fields

### PR DESCRIPTION
**Pull request by neozhaoliang**
related issue = #5420

Here is currently how we use `ti.append`:

```python
import taichi as ti
ti.init(arch=ti.cpu)

x = ti.field(int, shape=None)
q = ti.root.dynamic(ti.i, 5)
q.place(x)

@ti.kernel
def test():
    for i in range(5):
        ti.append(q, [], i)   # q = x.parent()
    print(ti.length(q, []))

test()
```
The `ti.append()` function is a bit confusing in that it requires two more positional arguments compared with the usual `append` method.

We hope to change it to something like `x[i].append(i)`, this is done in the ast part by replacing `x[indices].append(val)` with `ti.append(x.parent(), indices, val)`.

### CODE REVIEW

**Comment by neozhaoliang** [comment]
@lin-hitonami Thanks a lot for this review, it now works!

**Comment by neozhaoliang** [comment]
@yuanming-hu Could you please help explain the drawback of the hack in this PR? For example which scenario it would fail? Me and @lin-hitonami need more info here.

File: tests/python/test_dynamic_append_length.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def test_dynamic_append_length_f32():
+    _test_dynamic_append_length(ti.f32)
+
+
+@test_utils.test(exclude=[ti.cc, ti.opengl, ti.vulkan],
+                 default_fp=ti.f64,
+                 require=ti.extension.data64,
+                 debug=True)
+def test_dynamic_append_length_f64():
+    _test_dynamic_append_length(ti.f64)
```
Comment (yuanming-hu) [inline tests/python/test_dynamic_append_length.py:36]:
```suggestion
```

I don't think `ti.f64` works here - the test passes likely by luck. See 
https://github.com/taichi-dev/taichi/blob/5b69178e2ee45a2f2fe9ecea786bd61356f0a8d5/taichi/runtime/llvm/runtime_module/node_dynamic.h#L61

**Review by yuanming-hu — APPROVED** [review]
LGTM with one nit. Thanks!

**Pull request merged**